indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -125,10 +124,13 @@ public void run()
         log.info("Started WorkerTaskMonitor.");
         started = true;
       }
+      catch (InterruptedException e) {
+        throw e;
```

Comment:
Ok, I think all the interrupts in this PR are handled reasonably. This one is thrown out of lifecycle.start() without being caught.

The others seem properly thrown out of the notice handler loop.

Sorry, it just takes a little while to dig through interrupted exceptions to make sure they are handled reasonably